db/tables/operations/split.py

```diff
@@ -69,50 +58,43 @@ def _create_split_insert_stmt(old_table, extracted_table, extracted_columns, rem
         .returning(literal(1))
         .cte()
     )
-    remainder_sel = select(
-        cte_remainder_columns,
-        distinct=True
-    ).where(exists(extract_ins_cte.select()))
-
+    fk_update_dict = {remainder_fk_name: split_cte.c[SPLIT_ID]}
     split_ins = (
-        remainder_table
-        .insert()
-        .from_select(
-            [remainder_fk_name] + remainder_column_names,
-            remainder_sel
-        )
+        old_table
+        .update().values(**fk_update_dict).
+        where(old_table.c[constants.ID] == split_cte.c[constants.ID],
+              exists(extract_ins_cte.select()))
     )
     return split_ins
 
 
-def extract_columns_from_table(old_table_name, extracted_column_names, extracted_table_name, remainder_table_name, schema, engine, drop_original_table=False):
+def extract_columns_from_table(old_table_oid, extracted_column_attnums, extracted_table_name, schema, engine,):
+    old_table_name = reflect_table_from_oid(old_table_oid, engine).name
     old_table = reflect_table(old_table_name, schema, engine)
     old_columns = (MathesarColumn.from_column(col) for col in old_table.columns)
     old_non_default_columns = [
         col for col in old_columns if not col.is_default
     ]
+    extracted_column_names = get_columns_name_from_attnums(old_table_oid, extracted_column_attnums, engine)
     extracted_columns, remainder_columns = _split_column_list(
```

Comment:
`remainder_columns` is no longer accessed. I think you should consider removing the `_split_column_list` function, or simplifying it since there's no reason to create the list of remainder columns and pass it around with the new way of doing things.